### PR TITLE
Remove openvpn session manager

### PR DESCRIPTION
**Pull request by Waldz**
Updates #346

### CODE REVIEW

File: services/openvpn/session/validator_mocks.go [hunk 1]
```diff
@@ -45,7 +45,7 @@ func (manager *MockSessionManager) Create(peerID identity.Identity) (sessionInst
 	return session.Session{}, nil
 }
 
-// FindSession returns mocked session
+// FindClientSession returns mocked session
 func (manager *MockSessionManager) FindSession(sessionID session.SessionID) (session.Session, bool) {
```
Comment (soffokl) [inline services/openvpn/session/validator_mocks.go]: Comment and function name does not match.

File: session/manager.go [hunk 2]
```diff
@@ -58,7 +58,7 @@ func (manager *manager) Create(peerID identity.Identity) (sessionInstance Sessio
 	return sessionInstance, nil
 }
 
-// FindSession returns underlying session instance
+// FindClientSession returns underlying session instance
```
Comment (soffokl) [inline session/manager.go]: Comment and function name does not match.

File: session/manager_fake.go [hunk 3]
```diff
@@ -35,7 +35,7 @@ func (manager *ManagerFake) Create(peerID identity.Identity) (Session, error) {
 	return Session{"new-id", fakeConfig, peerID}, nil
 }
 
-// FindSession always returns empty session and signals that session is not found
+// FindClientSession always returns empty session and signals that session is not found
```
Comment (soffokl) [inline session/manager_fake.go]: Comment and function name does not match.

File: services/openvpn/session/validator_mocks.go (hunk 1 above)
Comment (Waldz) [reply services/openvpn/session/validator_mocks.go]: Fixed

File: session/manager.go (hunk 2 above)
Comment (Waldz) [reply session/manager.go]: Fixed

File: services/openvpn/session/client_map.go [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+type clientMap struct {
+	sessionManager   session.Manager
+	sessionClientIDs map[session.SessionID]int
+	sessionMapLock   sync.Mutex
+}
+
+// FindClientSession returns OpenVPN session instance by given session id
+func (cm *clientMap) FindClientSession(clientID int, id session.SessionID) (session.Session, bool, error) {
+	sessionInstance, foundSession := cm.sessionManager.FindSession(id)
```
Comment (vkuznecovas) [inline services/openvpn/session/client_map.go]: rename "foundSession" to "sessionExists"

File: services/openvpn/session/client_map.go [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+	if !foundSession {
+		return session.Session{}, false, errors.New("no underlying session exists, possible break-in attempt")
+	}
+
+	sessionClientID, clientIDFound := cm.sessionClientIDs[id]
```
Comment (vkuznecovas) [inline services/openvpn/session/client_map.go]: Decide on either clientIDFound or foundClientID, not both.

File: services/openvpn/session/client_map.go [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+	return sessionInstance, clientIDFound, nil
+}
+
+// UpdateClientSession updates OpenVPN session with clientID, returns false on clientID conflict
+func (cm *clientMap) UpdateClientSession(clientID int, id session.SessionID) bool {
+	cm.sessionMapLock.Lock()
+	defer cm.sessionMapLock.Unlock()
+
+	_, foundClientID := cm.sessionClientIDs[id]
```
Comment (vkuznecovas) [inline services/openvpn/session/client_map.go]: Decide on either clientIDFound or foundClientID, not both.

File: services/openvpn/session/client_map.go (hunk 4 above)
Comment (Waldz) [reply services/openvpn/session/client_map.go]: Renamed

File: services/openvpn/session/client_map.go (hunk 5 above)
Comment (Waldz) [reply services/openvpn/session/client_map.go]: Renamed

File: services/openvpn/session/client_map.go (hunk 6 above)
Comment (Waldz) [reply services/openvpn/session/client_map.go]: Renamed

**Pull request merged**